client/components/social-buttons/google.js

```diff
@@ -32,7 +34,9 @@ class GoogleLoginButton extends Component {
 	};
 
 	static defaultProps = {
-		scope: 'https://www.googleapis.com/auth/userinfo.profile',
+		scope: config.isEnabled( 'migration/sign-in-with-google' )
+			? 'openid profile email'
```

Comment:
The https://www.googleapis.com/auth/userinfo.profile scope does not include the e-mail, which leads to failures since the account creation endpoint requires the e-mail for identification purposes.